orders/parser.go

```diff
@@ -0,0 +1,65 @@
+package orders
+
+import (
+	"fmt"
+
+	dip "github.com/zond/godip/common"
+)
+
+type Parser struct {
+	generators []dip.Order
+}
+
+func NewParser(generators []dip.Order) Parser {
+	return Parser{generators}
+}
+
+func (self Parser) OrderTypes() (result []dip.OrderType) {
+	result = make([]dip.OrderType, len(self.generators))
+	for index, gen := range self.generators {
+		result[index] = gen.DisplayType()
+	}
+	return
+}
+
+func (self Parser) Orders() (result []dip.Order) {
+	result = make([]dip.Order, len(self.generators))
+	for index, gen := range self.generators {
```

Comment:
If you don't want this to return the actual prototype orders (which might be good to avoid bugs where they get mutated), use "copy" instead of the loop?